File: encodify/Classes/Encode/Controller/EncodeViewController.m
//
//  EncodeViewController.m
//  encodify
//
//  Created by 洪朔 on 16/8/31.
//  Copyright © 2016年 Tuccuay. All rights reserved.
//

#import "EncodeViewController.h"

#import "EncodifyBase64.h"
#import "EncodifyUnicode.h"
#import "EncodifyMorse.h"
#import "EncodifyURI.h"

@interface EncodeViewController ()

@end

@implementation EncodeViewController

- (NSString *)titleForPagerTabStripViewController:(XLPagerTabStripViewController *)pagerTabStripViewController {
    return @"Encode";
}

- (NSString *)encodeWithBase64:(NSString *)inputString {
    return [EncodifyBase64 encodeByBase64WithString:inputString];
}

- (NSString *)encodeWithUnicode:(NSString *)inputString {
    return [EncodifyUnicode encodeToUnicodeStringWithString:inputString];
}

- (NSString *)encodeWithMorse:(NSString *)inputString {
    return [EncodifyMorse encodeByMorseWithString:inputString];
}

- (void)encodeWithMorse:(NSString *)inputString textView:(UITextView *)textView {
    [EncodifyXmorseBridge encode:inputString complection:^(NSString *string) {
        textView.text = string;
    }];
}

- (NSString *)encodeWithURI:(NSString *)inputString {
    return [EncodifyURI encodeByURIWithString:inputString];
}

- (NSString *)encodeButtonTitle {
    return @"Encode";
}

@end

File: encodify/Classes/Encode/Controllers/EncodeViewController.swift
//
//  EncodeViewController.swift
//  encodify
//
//  Created by 洪朔 on 2024/12/20.
//  Copyright © 2024年 Tuccuay. All rights reserved.
//

import UIKit
import XLPagerTabStrip

class EncodeViewController: EncodeBaseViewController, IndicatorInfoProvider {
    
    // MARK: - IndicatorInfoProvider
    nonisolated func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo {
        return IndicatorInfo(title: "Encode")
    }
    
    // MARK: - Override encoding methods
    override func encodeWithBase64(_ inputString: String) -> String? {
        return Base64Encoder.encode(inputString)
    }
    
    override func encodeWithUnicode(_ inputString: String) -> String? {
        return UnicodeEncoder.encode(inputString)
    }
    
    override func encodeWithMorse(_ inputString: String) -> String? {
        return MorseEncoder.encode(inputString)
    }
    
    override func encodeWithURI(_ inputString: String) -> String? {
        return URLEncoder.encode(inputString)
    }
    
    override var encodeButtonTitle: String {
        return "Encode"
    }
}
